### PR TITLE
Migrate from CircleCI to GitHub Actions!

### PR DESCRIPTION
**Pull request by cahirwpz**
Since we're regularly running out of time on CircleCI it's high time to move to GH Actions but provide our own runner.

### CODE REVIEW

File: .github/workflows/default.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Mimiker
+
+on: [push, pull_request]
```
Comment (cahirwpz) [inline .github/workflows/default.yml:3]: I think we need to disable `pull_request` otherwise _GH Actions_ starts whole run twice.

Comment (pj1031999) [reply .github/workflows/default.yml]: done

File: .github/workflows/default.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      - uses: actions/checkout@v2
+      - run: make -j $(nproc) BOARD=rpi3 CLANG=1
+
+  kernel_tests_mips_gcc:
+    name: Tests MIPS - GCC with KASAN
+    runs-on: self-hosted
+    container: cahirwpz/mimiker-circleci:1.7.3
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
```
Comment (cahirwpz) [inline .github/workflows/default.yml:74]: You have an implicit dependency that fails if a runner doesn't pick up `build_mips_gcc` earlier.

File: .github/workflows/default.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          name: "mips gcc kernel+ramdisk"
+      - run: ./run_tests.py --board malta --thorough --non-interactive
+
+  kernel_tests_mips_clang:
+    name: Tests MIPS - Clang
+    runs-on: self-hosted
+    container: cahirwpz/mimiker-circleci:1.7.3
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
```
Comment (cahirwpz) [inline .github/workflows/default.yml:85]: Ditto.

**Review by cahirwpz — COMMENTED** [review]
I've launched four runners on `mimiker-ci`. That enabled job parallelism and caused problems with job dependencies that were not stated explicitly. Please fix that.

File: .github/workflows/default.yml (hunk 2 above)
Comment (cahirwpz) [reply .github/workflows/default.yml:79]: Done.

File: .github/workflows/default.yml (hunk 3 above)
Comment (cahirwpz) [reply .github/workflows/default.yml:91]: Done.

File: run_tests.py [hunk 4]
```diff
@@ -37,11 +36,11 @@ def gdb_inspect(interactive):
     gdb_cmd = getvar('gdb.binary')
     if interactive:
         gdb_opts = ['-iex=set auto-load safe-path {}/'.format(os.getcwd()),
-                    '-ex=target remote localhost:%d' % gdb_port(),
+                    '-ex=target remote localhost:%d' % RandomPort(),
```
Comment (cahirwpz) [inline run_tests.py:39]: I've a gut feeling it won't work for some reason... but we'll fix it later.

**Pull request merged**